core/range-controller.js

```diff
@@ -773,7 +773,13 @@ var RangeController = exports.RangeController = Montage.specialize( /** @lends R
                 // remove all values from the selection that were removed (but
                 // not added back)
                 minus.deleteEach(plus, equals);
-                if (this.selection) {
+
+                if (this.selection.length) {
+                    // ensure selection always has content
+                    if (this.content.length && this.avoidsEmptySelection && !this.multiSelect) {
+                        // selection can't contain previous content value as content already changed
+                        this.selection.add(this.content[this.content.length - 1]);
+                    }
```

Comment:
Ya I got that, but why the last one ? it seems just funny to me but it's a personal thought.